java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java

```diff
@@ -42,7 +42,7 @@
    * @param type
    * @return
    */
-  abstract protected FieldWriter getWriter(MinorType type);
+  abstract protected FieldWriter getWriter(ArrowType type);
```

Comment:
I think the only performance penalty is comparing an ArrowType instance instead of an enum compare - seems pretty minor.
Would you like me to split out into a separate PR?
MinorType is still used extensively in the writer framework - I think we ultimately need to pull it out since it doesn't capture the full arrow type. I noticed that the writers don't really handle decimal types, for that reason.
The other alternative is to create minor types for various fixed size lists - that would work for me, as I really only need 2-tuples at the moment. But it isn't especially flexible and @wesm asked for a more generic implementation.